lib/common.js

```diff
@@ -35,8 +37,8 @@ export default {
     },
 
     merge(src, dest) {
-        // NOTE: Do not merge arrays and only merge objects into objects.
-        if (!Thing.isArray(src) && Thing.isObject(src) && Thing.isObject(dest)) {
+        // NOTE: Do not merge arrays and only merge objects into objects. Do not merge special objects.
```

Comment:
What does it mean by special objects? can we clarify it more (with a bit of reasoning probably)? I can see there is check of getPrototypeof

Reply:
Updated the comment